### PR TITLE
Fix component info send along with GENERIC_ERROR

### PR DESCRIPTION
**Pull request by istoilovgithub**
In the case of command request timeout SDL responds properly, but without
component information. After the fix component information is added in the
form "'%component-name%' component does not respond"

### CODE REVIEW

**Comment by istoilovgithub** [comment]
Rebuild required

**Comment by istoilovgithub** [comment]
@okozlovlux Please review

File: src/components/application_manager/src/commands/command_request_impl.cc [hunk 1]
```diff
@@ -788,6 +790,78 @@ bool CommandRequestImpl::IsResultCodeUnsupported(
...
+  if (NsSmartDeviceLink::NsSmartObjects::invalid_object_value == app_type_so) {
+    LOG4CXX_WARN(logger_, "Invalid object for component type!");
+    return;
+  }
+
+  const int64_t app_hmi_type = app_type_so.asInt();
+  const std::string& component_name = AppHMITypeToString(
+      static_cast<mobile_apis::AppHMIType::eType>(app_hmi_type));
+  const std::string component_info =
+      component_name + " component does not respond";
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/src/commands/command_request_impl.cc]: @istoilovgithub please make description for function more clear. It is not clear from name and description this this function should be called only if some component does not respond.

File: src/components/application_manager/test/commands/command_request_impl_test.cc [hunk 2]
```diff
@@ -179,6 +179,9 @@ TEST_F(CommandRequestImplTest, OnTimeOut_StateAwaitingHMIResponse_SUCCESS) {
       app_mngr_,
       ManageMobileCommand(dummy_msg, Command::CommandOrigin::ORIGIN_SDL));
 
+  MockAppPtr app = CreateMockApp();
+  EXPECT_CALL(app_mngr_, application(_)).WillOnce(Return(app));
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/test/commands/command_request_impl_test.cc]: @istoilovgithub please use ON_CALL and specify app_id.

File: src/components/application_manager/test/commands/mobile/alert_request_test.cc [hunk 3]
```diff
@@ -224,6 +224,8 @@ TEST_F(AlertRequestTest, OnTimeout_GENERIC_ERROR) {
       ManageMobileCommand(_, am::commands::Command::CommandOrigin::ORIGIN_SDL))
       .WillOnce(DoAll(SaveArg<0>(&ui_command_result), Return(true)));
 
+  EXPECT_CALL(app_mngr_, application(_)).WillOnce(Return(mock_app_));
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/test/commands/mobile/alert_request_test.cc]: @istoilovgithub please use ON_CALL and specify app_id.

File: src/components/application_manager/test/commands/mobile/alert_request_test.cc [hunk 4]
```diff
@@ -319,6 +321,8 @@ TEST_F(AlertRequestTest, OnTimeOut_SUCCESS) {
       CreateNegativeResponse(_, _, _, mobile_apis::Result::GENERIC_ERROR))
       .WillOnce(Return(result_msg));
 
+  EXPECT_CALL(app_mngr_, application(_)).WillOnce(Return(mock_app_));
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/test/commands/mobile/alert_request_test.cc]: @istoilovgithub please use ON_CALL and specify app_id.

File: src/components/application_manager/src/commands/command_request_impl.cc (hunk 1 above)
Comment (istoilovgithub) [inline src/components/application_manager/src/commands/command_request_impl.cc]: @LuxoftAKutsan Fixed with 8402971

File: src/components/application_manager/test/commands/command_request_impl_test.cc (hunk 2 above)
Comment (istoilovgithub) [inline src/components/application_manager/test/commands/command_request_impl_test.cc]: @LuxoftAKutsan Fixed with 8402971

File: src/components/application_manager/test/commands/mobile/alert_request_test.cc (hunk 3 above)
Comment (istoilovgithub) [inline src/components/application_manager/test/commands/mobile/alert_request_test.cc]: @LuxoftAKutsan Fixed with 8402971

File: src/components/application_manager/test/commands/mobile/alert_request_test.cc (hunk 4 above)
Comment (istoilovgithub) [inline src/components/application_manager/test/commands/mobile/alert_request_test.cc]: @LuxoftAKutsan Fixed with 8402971

**Pull request merged**